sabnzbd/decoder.py

```diff
@@ -241,27 +247,31 @@ def decode_yenc(article: Article, raw_data: List[bytes]) -> bytes:
     # Let SABYenc do all the heavy lifting
     decoded_data, yenc_filename, crc_correct = sabyenc3.decode_usenet_chunks(raw_data)
 
-    # Mark as decoded
-    article.decoded = True
-
+    nzf = article.nzf
     # Assume it is yenc
-    article.nzf.type = "yenc"
+    nzf.type = "yenc"
 
     # Only set the name if it was found and not obfuscated
-    if not article.nzf.filename_checked and yenc_filename:
+    if not nzf.filename_checked and yenc_filename:
         # Set the md5-of-16k if this is the first article
         if article.lowest_partnum:
-            article.nzf.md5of16k = hashlib.md5(decoded_data[:16384]).digest()
+            nzf.md5of16k = hashlib.md5(decoded_data[:16384]).digest()
 
         # Try the rename, even if it's not the first article
         # For example when the first article was missing
-        article.nzf.nzo.verify_nzf_filename(article.nzf, yenc_filename)
+        nzf.nzo.verify_nzf_filename(nzf, yenc_filename)
 
     # CRC check
     if not crc_correct:
         logging.info("CRC Error in %s", article.article)
         raise BadData(decoded_data)
 
+    article.crc32 = crc32(decoded_data)
+    # Determine part size and precalculate crc_2pow
+    if article.lowest_partnum and len(decoded_data) > nzf.nzo.article_size:
+        nzf.nzo.article_size = len(decoded_data)
+        nzf.nzo.crc32_coeff = crc_2pow(nzf.nzo.article_size * 8)
```

Comment:
Should we maybe do it on a per-NZF basis? Then for season-packs there's a much higher success ratio.

Reply:
Considering that most nzbs are not season packs and a large share of the season packs will have content from mostly or only one poster, I think that the average number of CPU cycles used is lower when using a single precalculated value for the whole nzb. I could keep the nzo value and copy it to the nzf when it's the same but that seems overly complicated. Both are fast enough not to influence download speed for most users.